### PR TITLE
feat(policies): add Gr00t N1.7 policy

### PR DESCRIPTION
**Pull request by imstevenpmwork**
## Summary / Motivation

- Integrates NVIDIA Isaac-GR00T **N1.7** (Cosmos-Reason2/Qwen3-VL backbone + flow-matching action head, base model [`nvidia/GR00T-N1.7-3B`](https://huggingface.co/nvidia/GR00T-N1.7-3B)) as the `groot` policy type, replacing the N1.5 implementation. The new implementation is self-contained (`groot_n1_7.py`) and loads the backbone through `transformers>=5.4`, so the vendored N1.5 Eagle-2.5 model code can be deleted.
- Preprocessing, normalization, and the fine-tuning recipe are pinned to the native Isaac-GR00T contract and verified with a byte-level parity test against the original OSS implementation, so LeRobot fine-tuning/inference matches upstream behavior.
- Trade-off: this is a **breaking change** — N1.5 is no longer supported (see below). Dropping the in-repo Eagle backbone and the `flash-attn` build requirement makes `lerobot[groot]` installable again as part of `lerobot[all]`.

## Related issues

- Fixes / Closes: TBD close Gr00t N1.5 issues
- Related: Supersedes https://github.com/huggingface/lerobot/pull/3709

## What changed

- **New N1.7 model**: `src/lerobot/policies/groot/groot_n1_7.py` implements the GR00T N1.7 architecture (backbone loading, DiT action head, flow-matching sampler). Removed N1.5 code: `groot_n1.py`, `eagle2_hg_model/`, `action_head/flow_matching_action_head.py`, `action_head/action_encoder.py`.
- **Processor rewrite** (`processor_groot.py`): matches Isaac-GR00T preprocessing exactly — GPU/tensor image pipeline resizing to the trained resolution (256×256), train-time random crop (230×230) with eval center crop, opt-in letterboxing, per-embodiment modality configs read from the checkpoint's processor sidecar, and support for relative actions (RTC-only inference) including training-stats computation.
- **Training contract**: optimizer/scheduler/precision aligned with the native Isaac-GR00T fine-tuning recipe (AdamW lr=1e-4, betas=(0.9, 0.999), eps=1e-8, weight_decay=1e-5, grad clip 1.0), plus checkpoint-configured raw-state dropout during training.
- **Factory**: `make_pre_post_processors` now delegates to `make_groot_pre_post_processors_from_pretrained` (builds processors from the checkpoint sidecars), replacing the previous normalization-override patch.
- **N1.5 rejection**: N1.5 checkpoints, configs, and `model_version` spellings are detected and rejected with explicit migration guidance instead of failing obscurely.
- **Dependencies**: dropped `flash-attn` and `ninja` from the `groot` extra (no longer required), added `lerobot[dataset]` (needed for relative-action training stats); `lerobot[groot]` is re-enabled inside `lerobot[all]`.
- **Docs**: `docs/source/groot.mdx` and `docs/source/policy_groot_README.md` rewritten for N1.7 — SO-101 training command with image transforms, LIBERO training/eval commands and results, async-inference note (`inference.queue_threshold`) for stable real-robot control.

**Breaking change & migration**: GR00T N1.5 support is removed. To keep using an N1.5 checkpoint, pin the last release that supports it (`pip install 'lerobot==0.5.1'`); otherwise migrate to N1.7 (`--policy.base_model_path=nvidia/GR00T-N1.7-3B`).

## How was this tested (or how to run locally)

- Run locally: `uv run pytest tests/policies/groot -q` (the OSS parity and vs-original tests additionally need a GPU and the dumped artifacts).
- Manual checks: fine-tuned and evaluated on a real HW tasks (commands in `docs/source/groot.mdx`) and on LIBERO (results in the docs).

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [x] CI is green
- [x] Community Review: I have reviewed another contributor's open PR and linked it here: #

## Reviewer notes (items to be improved)
- `_runtime_dataset_meta` odd
- GPU utilization is low in training
- processor is complicated
- the policy has a dependecy on `LeRobotDataset` 
- add libero checkpoint and latest head results
- improvements on RTC side

### CODE REVIEW

File: src/lerobot/policies/groot/processor_groot.py [hunk 1]
```diff
@@ -1074,7 +1097,7 @@ def _build_n1_7_relative_action_processor_assets(
         if base_assets is not None
         else dict(N1_7_EMBODIMENT_MAPPING),
         formalize_language=base_assets.formalize_language if base_assets is not None else True,
-        valid_action_horizon=action_horizon,
+        valid_action_horizon=min(config.chunk_size, action_horizon),
```
Comment (acwrenn-nv) [inline src/lerobot/policies/groot/processor_groot.py:1110]: I think it seems reasonable to expect that chunk_size > action_horizon, does it make sense to have this?
Probably makes more sense to fail if thats true than to silent-clamp the valid action horizon?

Comment (acwrenn-nv) [reply src/lerobot/policies/groot/processor_groot.py:1110]: Well - we have that logic above, too, since action_horizon is already clamped.

**Comment by imstevenpmwork** [comment]
Full Test CI run: https://github.com/huggingface/lerobot/actions/runs/28678372264

**Comment by imstevenpmwork** [comment]
@claude Review this PR
